_posts/2016-10-26-jwt-artsy-journey.md

```diff
@@ -0,0 +1,46 @@
+---
+layout: post
+title: "JWT, Artsy Journey"
+date: 2016-10-26 11:03
+comments: true
+author: ashkan
+categories: [Auth, JWT, API]
+---
+
+At Artsy we currently have ~1730 client applications hitting our API and requesting authentication. On a successful authentication, we want to be able to provide basic user/application data in the token so clients can use it without needing to call our API. For that we want to use JWT.
```

Comment:
It's unclear whether you're talking about simply authenticating a client app or authenticating a user on behalf of a client app. Also, I think "without needing to call our API" is secondary to "without needing to look the session up in our database" (although the former is a real benefit of the switch we're currently navigating). Maybe: `When a user successfully authenticates through one of these clients, we want to embed basic user/application data in the resulting token rather than have to look up a session ID in the database on each request.`